build.cmd

```diff
@@ -485,7 +493,15 @@ if /i "%__DoCrossArchBuild%"=="1" (
     set __MsbuildWrn=/flp1:WarningsOnly;LogFile=!__BuildWrn!
     set __MsbuildErr=/flp2:ErrorsOnly;LogFile=!__BuildErr!
 
-    @call %__ProjectDir%\run.cmd build -Project=%__CrossCompIntermediatesDir%\install.vcxproj -configuration=%__BuildType% -platform=%__CrossArch% -MsBuildLog=!__MsbuildLog! -MsBuildWrn=!__MsbuildWrn! -MsBuildErr=!__MsbuildErr! %__RunArgs% %__UnprocessedBuildArgs%
+    if not defined NumberOfEnabledCore (
```

Comment:
you can make this a batch file subroutine (with `call`) to avoid repeating the code, or presumably just put it right at the top of the file so it always runs.

Reply:
Will move it into top of the file

Reply:
How is this different from using the predefined environment variable `NUMBER_OF_PROCESSORS` ?

Reply:
WMI which is used here provides several data points on machine: `NumberOfProcessors` which is equal to environment `NUMBER_OF_PROCESSORS` provides number of logical processors which comprises all threads available on processors implementing SMT. WMI property `NumberOfCores` provides physical core count of the CPU, whereas `NumberOfEnabledCore` property provides count of enabled cores on the CPU (some cores can be disabled via BIOS). Optimization runs indicated that number on which parallelisation should be based is a number of cores and not a number of threads and since some cores can be off it is necessary to retrieve enabled core count via WMI.